Tempo/Views/PostView.swift

```diff
@@ -102,18 +103,23 @@ class PostView: UIView, UIGestureRecognizerDelegate {
 				playNotificationHandler = NSNotificationCenter.defaultCenter().addObserverForName(PlayerDidChangeStateNotification,
 					object: post.player,
 					queue: NSOperationQueue.mainQueue(), usingBlock: { [weak self] note in
-					self?.updateProfileLabel()
-					self?.playerCellRef!.updatePlayingStatus()
-					self?.setUpTimer()
-					self?.setNeedsDisplay()
+						self?.updateProfileLabel()
+						self?.playerCellRef!.updatePlayingStatus()
+						self?.expandedPlayerRef!.updatePlayingStatus()
+						self?.setUpTimer()
+						self?.setNeedsDisplay()
 					})
  				
 				if let playerCellRef = playerCellRef {
-					likedNotificationHandler = NSNotificationCenter.defaultCenter().addObserverForName(PostLikedStatusChangeNotification,
-						object: playerCellRef,
-						queue: NSOperationQueue.mainQueue(), usingBlock: { [weak self] note in
+					likedNotificationHandler = NSNotificationCenter.defaultCenter().addObserverForName(PostLikedStatusChangeNotification, object: playerCellRef, queue: NSOperationQueue.mainQueue(), usingBlock: { [weak self] note in
 							self?.updateLikedStatus()
-						})
+					})
+				}
+				
+				if let expandedPlayerRef = expandedPlayerRef {
+					likedNotificationHandler = NSNotificationCenter.defaultCenter().addObserverForName(PostLikedStatusChangeNotification, object: expandedPlayerRef, queue: NSOperationQueue.mainQueue(), usingBlock: { [weak self] note in
```

Comment:
You better guarantee that the likedNotificationHandler isn't set here and above in this same call. Otherwise you're running into the issue I mentioned before.